pkg/adsc/adsc.go

```diff
@@ -952,3 +1074,54 @@ func (a *ADSC) GetEndpoints() map[string]*endpoint.ClusterLoadAssignment {
 	defer a.mutex.Unlock()
 	return a.eds
 }
+
+func (a *ADSC) handleMCP(gvk []string, rsc *any.Any, valBytes []byte) error {
+	if len(gvk) != 3 {
+		return nil // Not MCP
+	}
+	// Generic - fill up the store
+	if a.Store != nil {
+		m := &mcp.Resource{}
+		err := types.UnmarshalAny(&types.Any{
+			TypeUrl: rsc.TypeUrl,
+			Value:   rsc.Value,
+		}, m)
+		if err != nil {
+			return err
+		}
+		val, err := mcpToPilot(m)
+		if err != nil {
+			return err
+		}
+		val.GroupVersionKind = resource.GroupVersionKind{gvk[0], gvk[1], gvk[2]}
+		if err != nil {
+			adscLog.Warna("Invalid data ", err, " ", string(valBytes))
+		} else {
+			cfg := a.Store.Get(val.GroupVersionKind, val.Name, val.Namespace)
+			if cfg == nil {
+				_, err = a.Store.Create(*val)
+				if err != nil {
+					return err
+				}
+			} else {
+				_, err = a.Store.Update(*val)
+				if err != nil {
+					return err
+				}
+			}
+		}
+		if a.LocalCacheDir != "" {
+			strResponse, err := json.MarshalIndent(val, "  ", "  ")
+			if err != nil {
+				return err
+			}
+			err = ioutil.WriteFile(a.LocalCacheDir+"_res."+
```

Comment:
this isn't writing into local cache dir, you don't have a / after the dir

Reply:
It was intentional - probably should be called 'prefix'.

Ignore this part for now - it's mostly for debugging the code now. Eventually it should also read existing snapshot - then it'll be interesting, and I need to make the write match whatever we use in the 'file' source. Will adjust this code at that 
point. Can remove it from the PR as well - but it doesn't hurt much and keeping track of it is a pain...

It may become quite useful - the init container ( or even CNI when it grows some real go code ) could pre-fetch all the files, or maybe also an optional config map or volume with generated cached files from a CI/CD could be used so at 
startup and restart Envoy has a 'basic' working config, and only fetches changes.  Will improve startup time and resilience. But one step at a time - debug dump is the first.